src/priority_queue.py

```diff
@@ -0,0 +1,39 @@
+"""An implementation of a priority queue using a dictionary in Python."""
+
+
+class PriorityQueue(object):
+    """That's it."""
+
+    def __init__(self, value=None):
+        """Initialize a priority queue with default value as None."""
+        self._container = {}
+        if value and hasattr(value, "__iter__"):
+            for item in value:
+                self.insert(item[0], item[1])
+        elif value:
+            raise TypeError('Initial value must be an iterable!')
+
+    def insert(self, value, priority=0):
+        """Insert tuple in priority queue."""
+        if priority > 0:
+            raise ValueError('Priorities can only be negative, David.')
+        self._container.setdefault(priority, []).append(value)
+
+    def pop(self):
+        """Remove and return the highest priority value from the queue."""
+        try:
+            highest_priority = min([priority for priority in self._container])
+            highest_priority_value = self._container[highest_priority].pop(0)
```

Comment:
Note that popping from the front of a list can get computationally expensive for a large enough list.